flake8_pie/pie810_single_starts_ends_with.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import ast
+from functools import partial
+from typing import Any, Type
+
+from flake8_pie.base import Error
+
+
+def pie810_single_starts_ends_with(node: ast.BoolOp, errors: list[Error]) -> None:
+    """Check that multiple [starts/ends]with calls in an op operation do not look for
+    prefixes in the same string.
+    """
+    if not isinstance(node.op, ast.Or):
+        return
+
+    for method in ("startswith", "endswith"):
+        seen: dict[Type[ast.AST], Any] = {}
+        for val_node in node.values:
+            if (
+                isinstance(val_node, ast.Call)
+                and isinstance(val_node.func, ast.Attribute)
+                and val_node.func.attr == method
+                and isinstance(val_node.func.value, ast.Name)
+            ):
+                if val_node.func.value.id == "str":
+                    # str.startswith(stack, needle) -> stack
+                    arg = val_node.args[0]
+                else:
+                    # stack.startswith(needle) -> stack
+                    arg = val_node.func.value
+
+                if not isinstance(arg, (ast.Constant, ast.Name)):
+                    # we cannot check the equivalence of other types
+                    continue
+
+                t = type(arg)
+                val = arg.value if isinstance(arg, ast.Constant) else arg.id
+                seen.setdefault(t, set())
```

Comment:
We can use a defaultdict(set) to avoid the set default stuff but doesn’t matter — just a little cuter